marklogic-unit-test-modules/src/main/ml-modules/root/test/test-helper.xqy

```diff
@@ -754,14 +754,73 @@ declare function test:assert-false($conditions as xs:boolean*, $message as xs:st
     test:success()
 };
 
+(: Deprecated.  Replaced by test:assert-greater-than-or-equal(). :)
 declare function test:assert-meets-minimum-threshold($minimum as xs:decimal, $actual as xs:decimal+) {
-  test:assert-meets-minimum-threshold($minimum, $actual, ())
+  (
+    test:assert-meets-minimum-threshold($minimum, $actual, ()),
+    xdmp:log("Assertion 'assert-meets-minimum-threshold()' has been deprecated.   Use 'assert-greater-than-or-equal()' instead.", "info")
+  )
 };
 
+(: Deprecated.  Replaced by test:assert-greater-than-or-equal(). :)
 declare function test:assert-meets-minimum-threshold(
   $minimum as xs:decimal,
   $actual as xs:decimal+,
   $message as xs:string*
+) {
+  (
+    if (every $i in 1 to fn:count($actual) satisfies $actual[$i] ge $minimum) then
+      test:success()
+    else
+      let $message :=
+        fn:string-join((
+          $message,
+          "actual: " || xdmp:describe($actual) || " is less than minimum: " || xdmp:describe($minimum)
+        ), "; ")
+      return fn:error(xs:QName("ASSERT-MEETS-MINIMUM-THRESHOLD-FAILED"), $message, ($minimum, $actual))
+    ,
+    xdmp:log("Assertion 'assert-meets-minimum-threshold()' has been deprecated.   Use 'assert-greater-than-or-equal()' instead.", "info")
+  )
+};
+
+(: Deprecated.  Replaced by test:assert-less-than-or-equal(). :)
+declare function test:assert-meets-maximum-threshold($maximum as xs:decimal, $actual as xs:decimal+) {
+  (
+    test:assert-meets-maximum-threshold($maximum, $actual, ()),
+    xdmp:log("Assertion 'assert-meets-maximum-threshold()' has been deprecated.   Use 'assert-less-than-or-equal()' instead.", "info")
+  )
+};
+
+(: Deprecated.  Replaced by test:assert-less-than-or-equal(). :)
+declare function test:assert-meets-maximum-threshold(
+  $maximum as xs:decimal,
+  $actual as xs:decimal+,
+  $message as xs:string?
+) {
+  (
+    if (every $i in 1 to fn:count($actual) satisfies $actual[$i] le $maximum) then
+      test:success()
+    else
+      fn:error(
+        xs:QName("ASSERT-MEETS-MAXIMUM-THRESHOLD-FAILED"),
+        fn:string-join((
+          $message,
+          "actual: " || xdmp:describe($actual) || " is greater than maximum: " || xdmp:describe($maximum)
+        ), "; "),
+        ($maximum, $actual))
+    ,
+    xdmp:log("Assertion 'assert-meets-maximum-threshold()' has been deprecated.   Use 'assert-less-than-or-equal()' instead.", "info")
+  )
+};
+
+declare function test:assert-greater-than-or-equal($minimum as xs:decimal, $actual as xs:decimal+) {
+  test:assert-greater-than-or-equal($minimum, $actual, ())
+};
+
+declare function test:assert-greater-than-or-equal(
+  $minimum as xs:decimal,
```

Comment:
What are your thoughts on renaming `$maximum` to `$value`?

I think that convention is used in other frameworks.  See Hamcrest for example: https://github.com/hamcrest/JavaHamcrest/blob/5f54700d857619358c6467dc0e41e8d96ab4c235/hamcrest/src/main/java/org/hamcrest/Matchers.java#L1316

Reply:
Done.